lib/internal/modules/esm/loader.js

```diff
@@ -119,6 +119,7 @@ class DefaultModuleLoader {
       const { setCallbackForWrap } = require('internal/modules/esm/utils');
       const module = new ModuleWrap(url, undefined, source, 0, 0);
       setCallbackForWrap(module, {
+        initializeImportMeta: (meta, wrap) => this.importMetaInitialize(meta, { url }),
```

Comment:
Is it possible for eval to run multiple times? (I think yes) If so, I think we need to account for avoiding re-initialisation.

Also, shouldn't -e trigger the module translator ([where this already happens](https://github.com/nodejs/node/blob/5fa84e88661ac505ade404551bb5594e2201926e/lib/internal/modules/esm/translators.js#L118-L121))?

Reply:
If eval is run multiple time, it's going to create a different `module` object, which would have its own module context and would also need `import.meta`. If I'm wrong on that, can we please have a test that asserts that?

Reply:
Ah, yes. That makes sense.